### PR TITLE
lldp: Refactor parsing of received packet

### PR DESCRIPTION
**Pull request by fredrikdanielmoller**
Refactor pf_lldp_parse_packet() by moving code to separate sub-functions.
This reduces nesting and code size of pf_lldp_parse_packet(), making it easier to read.

Any change in functionality is unintentional.

Unit tests were also added.

### CODE REVIEW

File: src/common/pf_lldp.c [hunk 1]
```diff
@@ -1283,74 +1482,42 @@ int pf_lldp_parse_packet (
          break;
       case LLDP_TYPE_ORG_SPEC:
       {
-         uint8_t org_id[3];
-         uint8_t org_subtype;
-         pf_get_mem (&parse_info, &offset, sizeof (org_id), org_id);
-         org_subtype = pf_get_byte (&parse_info, &offset);
+         pf_lldp_org_header_t org;
```
Comment (pyhys) [inline src/common/pf_lldp.c:1485]: Typically we put these declaration in the beginning of the function in this stack.

**Review by pyhys — APPROVED** [review]

File: src/common/pf_lldp.c (hunk 1 above)
Comment (fredrikdanielmoller) [reply src/common/pf_lldp.c]: Done.

**Comment by pyhys** [comment]
Great!

**Review by olbjo — APPROVED** [review]
👍

**Review by hefloryd — APPROVED** [review]
Needs rebasing.

**Comment by fredrikdanielmoller** [comment]
Rebase and reformat

**Pull request merged**